tests/run-pass/threadleak_ignored.rs

```diff
@@ -0,0 +1,25 @@
+// ignore-windows: Concurrency on Windows is not supported yet.
+// compile-flags: -Zmiri-ignore-leaks
+
+//! Test that leaking threads works, and that their destructors are not executed.
+
+use std::cell::RefCell;
+
+struct LoudDrop(i32);
+impl Drop for LoudDrop {
+    fn drop(&mut self) {
+        eprintln!("Dropping {}", self.0);
+    }
+}
+
+thread_local! {
+    static X: RefCell<Option<LoudDrop>> = RefCell::new(None);
+}
+
+fn main() {
+    X.with(|x| *x.borrow_mut() = Some(LoudDrop(0)));
+    
+    let _detached = std::thread::spawn(|| {
+        X.with(|x| *x.borrow_mut() = Some(LoudDrop(1)));
```

Comment:
shouldn't this go into a loop after the assignment? Otherwise I think this test depends on the thread scheduling strategy. Or are TLS destructors only called on `join` not if the thread terminates by itself?

Reply:
TLS destructors are called as soon as the thread terminates. They are called by the terminating thread.

Reply:
Hm yeah... with an infinite loop it also becomes blindingly obvious that we cannot possibly run those destructors, as the TLS vars might be currently used.^^